testutil/accumulator.go

```diff
@@ -161,6 +161,17 @@ func (a *Accumulator) Get(measurement string) (*Metric, bool) {
 	return nil, false
 }
 
+// Calls the given Gather function and returns the first error found.
+func (a *Accumulator) GatherError(gf func(telegraf.Accumulator) error) error {
+	if err := gf(a); err != nil {
+		return err
+	}
+	if len(a.Errors) > 0 {
+		return a.Errors[0]
+	}
+	return nil
+}
```

Comment:
I proposed this a few times, but was always declined by your predecessor. Not sure what the reasons were though.
https://github.com/influxdata/telegraf/issues/1446#issuecomment-233371520
https://github.com/influxdata/telegraf/issues/2326#issuecomment-277557824